### PR TITLE
Additional fields

### PR DESCRIPTION
**Pull request by jjzieve**
This is a rebase from develop and some tweaks to @brianhill11's PR (#20).
Tweaks include:
- Addressing Ryan's comments from the original PR
- Adding a regression test for it.
- Adding support for indels in BAlleleFreqFormat
- Updating the README

### CODE REVIEW

**Comment by jjzieve** [comment]
@KelleyRyanM ignore my previous comment, I see some of the indel logic in GenotypeFormat, I think I can apply that

File: BAlleleFreqFormat.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
+from vcf.parser import _Format
+from IlluminaBeadArrayFiles import RefStrand
+import numpy as np
+
+REVERSE_COMPLIMENT = {"A": "T", "T": "A", "G": "C", "C": "G", "I": "I", "D": "D"}
```
Comment (KelleyRyanM) [inline BAlleleFreqFormat.py]: typo

File: BAlleleFreqFormat.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    Splits SNP string into tuple of individual alleles
+
+    Args: snp_string (string): allele string of the format
+        [T/C] or [G/C]
+
+    Returns:
+         allele1 (string), allele2 (string)
+    """
+    (allele1, allele2) = snp_string[1:4].split("/")
+    assert allele1 in "ATGCID", "allele %r is invalid (expected A,T,G,C,I,D)" % allele1
```
Comment (KelleyRyanM) [inline BAlleleFreqFormat.py]: Should we just check if key in REVERSE_COMPLEMENT?

File: BAlleleFreqFormat.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    assert allele2 in "ATGCID", "allele %r is invalid (expected A,T,G,C,I,D)" % allele2
+    return allele1, allele2
+
+
+def normalize_alleles_by_strand(snp_string):
```
Comment (KelleyRyanM) [inline BAlleleFreqFormat.py]: Name here is misleading, should probably just indicate that it extract the reverse complement

**Comment by KelleyRyanM** [comment]
@jjzieve ,
  Have you or @KyleLevi verified this pull request is still compatible with Python 3? 
  -Ryan

**Comment by jjzieve** [comment]
@KelleyRyanM Nope, I can check python3 compatibility and address the other comments

**Comment by jjzieve** [comment]
@KelleyRyanM Ran regressions on python 3.5.6 and it worked. Seemed that was the highest version that pysam 0.9.0 had been built on.

**Comment by jjzieve** [comment]
@KelleyRyanM Kyle had failed regressions due to precision differences for GQ and LRR with an updated numpy module. I've confirmed with numpy=1.15.2 (I believe he has 1.16.3).
_i.e._
regression ->`0/0:**0.437028**:0.0:**0.167821**`
updated numpy -> `0/0:**0.43702784**:0.0:**0.16782084**`

Is this acceptable since the README points to one version (1.11)?
If not, what's the best way to handle this? This SO post seems to outline the problem:
https://stackoverflow.com/questions/4844865/how-to-set-the-precision-on-strnumpy-float64
I'm guessing numpy updated the default precision from 6 to 8 or something somewhere between 1.11 and 1.15

**Comment by KelleyRyanM** [comment]
@jjzieve ,
  Can't we just format the string when returning from the BAlleleFreq formatter (e.g., return value -> return %.6f' % value)?

**Comment by jjzieve** [comment]
@KelleyRyanM yes, but its impacting gencall score also, it looks like it can't easily be fixed without re-generating the truth vcfs for the regressions. Also it appears round(value, 6) might be better than format since it doesn't pad 0.0 to 0.000000 (though maybe we want that?). But I can re-gen the vcfs and see how many values change. Most of the gencalls are 6 decimal places but I see some that are 5, 7, etc.

**Comment by jjzieve** [comment]
@KelleyRyanM and @KyleLevi it impacts 30+ loci. Maybe I can move that effort to a separate PR/issue? There could be some usefulness in providing the option to specify the precision of the output? But I think that's a bit out of scope for this PR.

**Comment by jjzieve** [comment]
Spoke to Ryan, opening a seperate issue for the precision differences with numpy updates.

**Pull request merged**